dts/bindings/spi/ambiq,spid.yaml

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2024 Ambiq <www.ambiq.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Ambiq SPI Device
+
+compatible: "ambiq,spid"
+
+include: [spi-controller.yaml, pinctrl-device.yaml, ambiq-pwrcfg.yaml]
```

Comment:
Should the ambiq,spid.yaml have the spi-controller.yaml or the spi-device.yaml?

Reply:
I created this by reference to what nrf-spis.yaml includes; and changing to spi-device.yaml will cause lots of builld errors, I guess the spi-device.yaml is used for the external device connected to the soc, but not for the soc itself works in device mode.